### DISCUSSION TITLE
pip install ffcv does not work on google colab! any idea?

### DISCUSSION THREAD

**Discussion by sabeesh90** [libffcv/ffcv#55] [Q&A]
Collecting ffcv
  Using cached ffcv-0.0.2.tar.gz (53 kB)
WARNING: Discarding https://files.pythonhosted.org/packages/37/5a/15f9a2f88c07d41a1830e55b133bb2f96bb3a5af77d125d35bed3f5c6535/ffcv-0.0.2.tar.gz#sha256=549c78dce66edc825118bab640eac9032417bcd8a06609b9eaf2d22ef8e8a52e (from https://pypi.org/simple/ffcv/). Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.
  Using cached ffcv-0.0.1.tar.gz (49 kB)
WARNING: Discarding https://files.pythonhosted.org/packages/4f/55/9b06a72c29710110387c3af33eb1a9d6e5d9d5781d5b2850a3ad202942d2/ffcv-0.0.1.tar.gz#sha256=5246dbbbc0a3bcf788783d413292d8bede455a08ad0fa9c29736eca85b577b26 (from https://pypi.org/simple/ffcv/). Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.

ERROR: Could not find a version that satisfies the requirement ffcv (from versions: 0.0.1, 0.0.2)
ERROR: No matching distribution found for ffcv

**Comment by Hadisalman** [comment]
Hi @sabeesh90! We are preparing a demo of ffcv in Goolge Colab. Stay tuned!

**Reply by sabeesh90** [reply]
That's amazing! looking forward to use your library in one of our researches! thanks. Im waiting!

**Comment by GuillaumeLeclerc** [comment]
Hello @sabeesh90. We tried our best but it seems that Google Colab only support python 3.7.1 and FFCV relies heavily on the standard library called `processing.shared_memory` which is only available starting in python 3.8.0. So until Google decides to update their 4 year old python there is nothing we can do. And believe me we really tried our absolute best to have python 3.8.0 running but the only thing we could achieve is crashing the user interface.

**Comment by bibhabasumohapatra** [comment]
off topic . . .but does it work on kaggle?
can you suggest . . .
installation through github? 
or git clone . . . way of doing things?

**Reply by GuillaumeLeclerc** [reply]
I have no experience with Kaggle notebooks. What are you allowed to do there? are you root? What version of python does it come with ? Why not using the recommended way to install FFCV ?

**Reply by bibhabasumohapatra** [reply]
no . . . IG there is no way upgrading python from  3.7.1(current python in kaggle) to 3.8
btw . . . due to some internet issues(HTTP errors) I was facing problems installing torch 1.10.2 (1.5 GB) , So i had already installed version of that in one of my envs, So thought using the other way of installing in that env would skip my install problems.

**Reply by GuillaumeLeclerc** [reply]
FFCV requires python 3.8. Nothing I can do about that sorry

**Reply by bibhabasumohapatra** [reply]
@GuillaumeLeclerc 
![image](https://user-images.githubusercontent.com/68384968/153897900-d4dc55eb-9b6e-4083-8aa2-e0cc1f89ff88.png)

![image](https://user-images.githubusercontent.com/68384968/153896969-f2220c0d-ea1c-4ca9-8201-4f87c8ac68e8.png)
I have some how found way . . . for python 3.9 on colab, anyway after that any steps to follow on colab?

**Reply by GuillaumeLeclerc** [reply]
@bibhabasumohapatra , this doesn't work. It does make the version of the program `python` but that's not the program actually used by Collab to run your cells. Try to import a 3.8 specific feature and you will see it fail.

We tried much more complicated approaches. I managed to actually change the version of what is used by Collab to run your cells but they speak a different protocol so the website is confused and crashs until you wipe your environment. We went and talked to Colab developers and they said it was impossible so I don't think there is hope before they actually upgrade python themselves (including the user interface)
